lib/streamlit/cli.py

```diff
@@ -118,17 +120,74 @@ def main_hello():
     _main_run(filename)
 
 
+def _compose_option_parameter(config_option):
+    """
+    Composes given config option options as options for click lib.
+    """
+    option = "--{}".format(config_option.key)
+    param = config_option.key.replace(".", "_")
+    description = config_option.description
+    if config_option.deprecated:
+        description += "\n {} - {}".format(
+            config_option.deprecation_text, config_option.deprecation_date
+        )
+
+    return {
+        "param": param,
+        "description": description,
+        "type": config_option.type,
+        "option": option,
+    }
+
+
+def configurator_options(func):
+    """
+    Decorator that adds config param keys to click dynamically.
+    """
+    for _, value in reversed(_config._config_options.items()):
```

Comment:
Why do you need `reversed` here?

Reply:
for some reason the default order gives me the last config item first. I don't know why I just saw it was this way and reversed it.

Reply:
Oh actually it makes sense now. `click` probably reverses them internally so the help string appears in the same order as you declare the decorators. The topmost decorator should show up first on the help.

But the topmost decorator is actually the _last_ decorator that is applied. So on our case, we should reverse again to make sure the thing we want to have at the top is applied last.